packages/serverpod/lib/src/database/database_result.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:meta/meta.dart';
+import 'package:serverpod/database.dart';
+
+@internal
+
+/// Prepares a query result for serverpod serialization.
+Map<String, dynamic>? resolvePrefixedQueryRow(
+    Table table, Map<String, Map<String, dynamic>> rawRow,
+    {Include? include, bool lookupWithColumnName = false}) {
+  // Resolve this object.
+  var rawTableRow = rawRow[table.tableName];
+  if (rawTableRow == null) return null;
+
+  var resolvedTableRow = _createColumnMapFromPrefixedColumns(
+    table.columns,
+    rawTableRow,
+    lookupWithColumnName,
+  );
+
+  if (resolvedTableRow.isEmpty) {
+    return null;
+  }
+
+  // Resolve all includes for the object.
+  include?.includes.forEach((relationField, relationInclude) {
+    if (relationInclude == null) return;
+
+    var relationTable = table.getRelationTable(relationField);
+    if (relationTable == null) return;
+
+    resolvedTableRow[relationField] = resolvePrefixedQueryRow(
+      relationTable,
+      rawRow,
+      include: relationInclude,
+    );
+  });
+
+  return resolvedTableRow;
+}
+
+Map<String, dynamic> _createColumnMapFromPrefixedColumns(
+  List<Column> columns,
+  Map<String, dynamic> rawTableRow,
+  lookupWithColumnName,
```

Comment:
Narrow down the scope of SelectQueryBuilder so that it always required columns as fields so that we can remove this bool. (Always fetch with queryAlias)